internal/pkg/service/common/servicectx/servicectx.go

```diff
@@ -200,7 +203,7 @@ func (v *Process) Add(operation func(ShutdownFn)) {
 
 	select {
 	case <-v.terminating:
-		v.logger.Errorf(`cannot Add operation: the Process is terminating`)
+		v.logger.ErrorfCtx(v.shutdownCtx, `cannot Add operation: the Process is terminating`)
```

Comment:
Two log message are problematic, there are 2 possible solutions:
1. Add a `ctx` argument to the `Add` and `OnShutdown` method, so the message will be logged in the parent context.
2. Store `shutdown` context and use it here - it is anti-pattern and we try to avoid it.
3. Use `context.Background()` - it is an unexpected state, but we dont want to stop shutdown via `panic`.   
  So we want DataDog to generate an alert that an error has occurred, we don't need more.

------------

I asked myself **where I want to see this log message?**

And it makes sense to me to see that something unexpected happens during the shutdown operation (sleep), so I prefered solution `2.` (or `3.`).